packages/flutter/lib/src/widgets/dismissable.dart

```diff
@@ -249,31 +263,51 @@ class _DismissableState extends State<Dismissable> {
   }
 
   Widget build(BuildContext context) {
+    Widget background;
+    if (config.backgrounds != null)
+      background = config.backgrounds[_dismissDirection] ?? config.backgrounds[config.direction];
+
     if (_resizeAnimation != null) {
       // we've been dragged aside, and are now resizing.
       assert(() {
         if (_resizeAnimation.status != AnimationStatus.forward) {
           assert(_resizeAnimation.status == AnimationStatus.completed);
           throw new WidgetError(
-            'Dismissable widget completed its resize animation without being removed from the tree.\n'
-            'Make sure to implement the onDismissed handler and to immediately remove the Dismissable '
+            'A dismissed Dismissable widget is still part of the tree.\n' +
+            (config.key != null ? '' : "Dismissable widgets must specifiy a key but this widget's key is null.\n") +
```

Comment:
Discovering and explaining that Dismissable widgets in structurally symmetric locations (or whatever) need distinguishing keys is too hard. I'll make the key a required parameter.